src/wireviz/build_examples.py

```diff
@@ -1,113 +1,149 @@
+                            out.write('\n\n')
+                    else:
+                        out.write(f'## Example {i}\n')
+
+                    if include_source:
+                        with open_file_read(yaml_file) as src:
+                            out.write('```yaml\n')
+                            for line in src:
+                                out.write(line)
+                            out.write('```\n')
+                        out.write('\n')
+
+                    out.write(f'![]({yaml_file.stem}.png)\n\n')
+                    out.write(f'[Source]({yaml_file.name}) - [Bill of Materials]({yaml_file.stem}.bom.tsv)\n\n\n')
+
+
+def clean_generated(groupkeys):
+    for key in groupkeys:
+        # collect and remove files
+        for filename in collect_filenames('Cleaning', key, generated_extensions):
+            if filename.is_file():
+                print(f'  rm {filename}')
+                os.remove(filename)
+
+
+def compare_generated(groupkeys, include_from_graphviz = False):
+    compare_extensions = generated_extensions if include_from_graphviz else extensions_not_from_graphviz
+    for key in groupkeys:
+        # collect and compare files
+        for filename in collect_filenames('Comparing', key, compare_extensions):
+            cmd = f'git --no-pager diff {filename}'
+            print(f'  {cmd}')
+            os.system(cmd)
+
+
+def restore_generated(groupkeys):
+    for key in groupkeys:
+        # collect input YAML files
+        filename_list = collect_filenames('Restoring', key, input_extensions)
+        # collect files to restore
+        filename_list = [fn.with_suffix(ext) for fn in filename_list for ext in generated_extensions]
+        if readme in groups[key]:
+            filename_list.append(groups[key]['path'] / readme)
+        # restore files
+        for filename in filename_list:
+            cmd = f'git checkout -- {filename}'
+            print(f'  {cmd}')
+            os.system(cmd)
 
 
 def parse_args():
-    parser = argparse.ArgumentParser(
-        description='Wireviz Example Manager',
-    )
-    parser.add_argument('action', nargs='?', action='store', default='build')
-    parser.add_argument('-generate', nargs='*', choices=['examples', 'demos', 'tutorials'], default=['examples', 'demos', 'tutorials'])
+    parser = argparse.ArgumentParser(description='Wireviz Example Manager',)
+    parser.add_argument('action', nargs='?', action='store',
+                        choices=['build','clean','compare','restore'], default='build',
+                        help='what to do with the generated files (default: build)')
+    parser.add_argument('-c', '--compare-graphviz-output', action='store_true',
+                        help='the Graphviz output is also compared (default: False)')
+    parser.add_argument('-g', '--groups', nargs='+',
+                        choices=groups.keys(), default=groups.keys(),
+                        help='the groups of generated files (default: all)')
```

Comment:
This is by no means mandatory, but if we're using the `Click` package for CLI parsing in `wireviz.py`, maybe it makes sense to use it here too, for consistency?

Reply:
I agree, that's why I wrote in the PR description:
> - Run python build_examples.py -h or with --help to print the generated help. The auto-generated help text is not the best, but better than maintaining it manually. I wonder if click as recommended in #60 (comment) could improve this?

I will have a look at that.

Reply:
I don't think I have the time to complete this today. Maybe it's best to finish this PR without click?

I have looked into click, but I didn't find a way to implement the `--groups` option. The documentation not very detailed. Here is what I have tried so far, but I also got a lot of errors (e.g. `SyntaxError: invalid syntax`) I couldn't figure out right now:
```diff
----------------------------------- setup.py -----------------------------------
index 09621bb..cc5a3da 100644
@@ -24,6 +24,7 @@ setup(
     install_requires=[
         'pyyaml',
         'graphviz',
+        'click',
         ],
     license='GPLv3',
     keywords='cable connector hardware harness wiring wiring-diagram wiring-harness',
@@ -31,7 +32,10 @@ setup(
     package_dir={'': 'src'},
     packages=find_packages('src'),
     entry_points={
-        'console_scripts': ['wireviz=wireviz.wireviz:main'],
+        'console_scripts': [
+            'wireviz=wireviz.wireviz:main',
+            'build_examples=wireviz.build_examples:cli',
+            ],
         },
     classifiers=[
         'Development Status :: 4 - Beta',

------------------------ src/wireviz/build_examples.py ------------------------
index 10a4b1b..97c2726 100755
@@ -4,6 +4,7 @@
 import argparse
 import sys
 import os
+import click
 from pathlib import Path
 
 script_path = Path(__file__).absolute()
@@ -48,6 +49,12 @@ def collect_filenames(description, groupkey, ext_list):
     return sorted([filename for pattern in patterns for filename in path.glob(pattern)])
 
 
+@click.group()
+cli()
+    pass
+
+
+@cli.command()
 def build_generated(groupkeys):
     for key in groupkeys:
         # preparation
@@ -87,6 +94,7 @@ def build_generated(groupkeys):
                     out.write(f'[Source]({yaml_file.name}) - [Bill of Materials]({yaml_file.stem}.bom.tsv)\n\n\n')
 
 
+@cli.command()
 def clean_generated(groupkeys):
     for key in groupkeys:
         # collect and remove files
@@ -96,6 +104,7 @@ def clean_generated(groupkeys):
                 os.remove(filename)
 
 
+@cli.command()
 def compare_generated(groupkeys, include_graphviz_output = False):
     compare_extensions = generated_extensions if include_graphviz_output else extensions_not_containing_graphviz_output
     for key in groupkeys:
@@ -106,6 +115,7 @@ def compare_generated(groupkeys, include_graphviz_output = False):
             os.system(cmd)
 
 
+@cli.command()
 def restore_generated(groupkeys):
     for key in groupkeys:
         # collect input YAML files
@@ -147,4 +157,4 @@ def main():
 
 
 if __name__ == '__main__':
-    main()
+    cli()
```

Reply:
Perhaps the [discussion](https://github.com/formatc1702/WireViz/pull/123#issuecomment-663593060) on #123 might help you. It seems `click` requires one `-g` for each group you want to include. Not sure if it's a limitation, or a feature, and what reason it might have.